### PR TITLE
WIP: custom nginx port via env variable

### PR DESCRIPTION
**Pull request by empeje**
The number of changed files should be divisible by 4 since we have 4 version (x86 and arm for each Alpine and Debian) + 1 for readme 😉 .

It is an update for automatically detect nginx port

Solved #29

### CODE REVIEW

**Review by Dashlorde — APPROVED** [review]

**Comment by empeje** [comment]
Upps, I found bug in alpine version. Please don't merge it yet, the start.sh need to be udpdated.

File: 3.4/arm_alpine/start.sh [hunk 1]
```diff
@@ -42,6 +42,8 @@ then
   echo "env[MOODOLE_DB_PORT] = '$MOODOLE_DB_PORT'" >> $PHP_FM_CONFIG
 fi
 
+envsubst '$NGINX_PORT' < $NGINX_CONFIG.template > $NGINX_CONFIG
```
Comment (empeje) [inline 3.4/arm_alpine/start.sh]: you should put it in the top in every version because we need to render the template before adding manipulation

**Comment by empeje** [comment]
Done solving the render template problem.

**Comment by xyb994** [comment]
@mappuji I tried with the following docker-compose file on Raspberry Pi and it worked fine. Do we need to change both `ports:` and `NGINX_PORT ` part? What's the right way to serve moodle at port 8080?

```
version: '2'
services:
  moodledb_rpi:
    image: arm32v7/postgres
    container_name: moodledb_rpi
    environment:
    # MAKE SURE THIS ONE SAME WITH THE MOODLE
    - POSTGRES_DATABASE=moodle
    - POSTGRES_USER=moodle
    - POSTGRES_PASSWORD=moodle
  moodle_rpi:
    image: treehouses/moodle:rpi-alpine-0.0.2-nginx-73baae59
    container_name: moodle_rpi
    ports:
      - "8080:8080"
    environment:
    - NGINX_PORT=8080
    - MOODOLE_DB_URL=moodledb_rpi
    - MOODOLE_DB_NAME=moodle
    - MOODOLE_DB_USER=moodle
    - MOODOLE_DB_PASS=moodle
    - MOODOLE_DB_PORT=5432
    - MOODOLE_MAX_BODY_SIZE=200M
    - MOODOLE_BODY_TIMEOUT=300s
```

**Comment by empeje** [comment]
Yes, you nailed it @xyb994

**Comment by dogi** [comment]
@xyb994 congrats - looks like we nailed it ...

... but I am very confused @mappuji 

and how do we convey to the users that they have to change all 3 numbers

**Comment by empeje** [comment]
Let me explain, this part

```
    - NGINX_PORT=8080
```

is actually doing the job to change the port. `NGINX_PORT` if leave empty will fallback to `80`


but, this part 

```
    ports:
      - "8080:8080"
```

is a way we do it in docker compose, we redirect (bind) port 8080 in container to port 8080 in the host machine.

**Comment by empeje** [comment]
What do you think @dogi ? Should I merge it or revise it?

**Comment by empeje** [comment]
More info about port here https://docs.docker.com/compose/compose-file/compose-file-v2/#ports

**Comment by empeje** [comment]
It is a bad idea to have a custom port, the better approach is bind the nginx server to `0.0.0.0` then redirect using docker compose feature.

**Comment by empeje** [comment]
After trial and error with #39

**Pull request merged**